### PR TITLE
Demote mocha-test-setup to fluid-internal package scope

### PR DESCRIPTION
**Pull request by ChumpChief**
The mocha-test-setup package isn't designed for use outside the FluidFramework repo, this change makes that explicit by moving it to the `@fluid-internal` package scope.  It can't be made fully private as it's used across release groups (by tinylicious, api-markdown-documenter, and benchmark), and for the same reason those references can't be updated until we release a version under the new scope.

This PR also drops the type testing for the package, which seems generally unnecessary anyway.

[AB#7294](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7294)

### CODE REVIEW

File: packages/test/mocha-test-setup/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# @fluidframework/mocha-test-setup
+# @fluid-internal/mocha-test-setup
```
Comment (Josmithr) [inline packages/test/mocha-test-setup/README.md:2]: While we're here...

```suggestion

<!-- AUTO-GENERATED-CONTENT:START (README_PACKAGE_SCOPE_NOTICE:packageJsonPath=./package.json) --><!-- AUTO-GENERATED-CONTENT:END -->

```

That way it's a bit more obvious to folks looking at the package on npm or wherever that this isn't intended for external use.

**Review by Josmithr — APPROVED** [review]
Thanks for doing this!

File: packages/test/mocha-test-setup/README.md (hunk 1 above)
Comment (ChumpChief) [reply packages/test/mocha-test-setup/README.md:2]: Very cool, I didn't even know about this as an option!

**Comment by ChumpChief** [comment]
@Abe27342 BTW this requires a merge-tree codeowner review, hoping you might be able to fill that role :)

File: packages/test/mocha-test-setup/README.md (hunk 1 above)
Comment (Josmithr) [reply packages/test/mocha-test-setup/README.md:2]: We've got a fair number of useful pragmas. Most of them are documented here: https://github.com/microsoft/FluidFramework/blob/main/tools/markdown-magic/README.md#transforms (though, notably, not the one I recommended here... I'll fix that...).

We'll want to update our namespace definitions for these if/when we update them.

**Comment by anthony-murphy** [comment]
> @Abe27342 BTW this requires a merge-tree codeowner review, hoping you might be able to fill that role :)

i don't see why it needs it, as it should only be for changes under /packages/dds/merge-tree/src, which i don't see here

File: packages/dds/sequence/src/test/memory/.mocharc.cjs [hunk 2]
```diff
@@ -14,7 +14,7 @@ module.exports = {
 	"recursive": true,
 	"reporter": "@fluid-tools/benchmark/dist/MochaMemoryTestReporter.js",
 	"reporterOptions": ["reportDir=.memoryTestsOutput/"],
-	"require": ["node_modules/@fluidframework/mocha-test-setup"],
+	"require": ["node_modules/@fluid-internal/mocha-test-setup"],
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/test/memory/.mocharc.cjs:17]: ahh found why it needs merge-tree cr

**Review by anthony-murphy — APPROVED** [review]

File: packages/test/mocha-test-setup/README.md (hunk 1 above)
Comment (Josmithr) [reply packages/test/mocha-test-setup/README.md:2]: #19763

**Comment by alexvy86** [comment]
> It can't be made fully private as it's used across release groups (by tinylicious, api-markdown-documenter, and benchmark)

Here's another candidate to use `file:` dependencies within the repo. Tinylicious would be an issue for that because of the docker build server does and not having access to all the repo by default, but if we get off that docker build at some point, I don't think the cross-release group use of mocha-test-setup should be a blocker to make it @fluid-private.

**Pull request merged**